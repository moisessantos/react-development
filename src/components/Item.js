import React, { Component } from 'react';

class Item extends Component {
  render() {
    return (
        <li>
            { this.props.children }
        </li>
    );
  }
}

export default Item;
