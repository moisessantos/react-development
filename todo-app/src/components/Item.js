import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Item extends Component {
  constructor(props) {
    super(props)
    this.state = {
      currentItems: props.items,
    };

    this.onDelete = this.onDelete.bind(this);

    this.ref = React.createRef();
  }

  render() {
    return (
        <li>
            <span ref={this.ref}>{ this.props.children } </span>
            <button onClick={this.onDelete}>Delete</button>
        </li>
    );
  }
  onDelete(){
    // this regex deletes trailing spaces at the end
    this.props.onDelete(this.ref.current.innerText.replace(/\s*$/,''));
  }
}

Item.propTypes = {
  onDelete: PropTypes.func.isRequired
};
export default Item;
