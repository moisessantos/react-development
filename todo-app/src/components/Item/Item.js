import React, { memo } from 'react';
import PropTypes from 'prop-types';

const Item = ({ children, handleClick }) => (
        <li className="list-group-item list-group-item-action">
            <span>{ children }</span>
            <button
              className="btn btn-danger" 
              onClick={(e) => handleClick(e.target.previousSibling.innerHTML)}
            >Delete</button>
        </li>
    );

Item.propTypes = {
  handleClick: PropTypes.func.isRequired
};

Item.displayName = 'Item';

export default memo(Item);
