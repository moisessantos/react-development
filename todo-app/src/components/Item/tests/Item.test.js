import React from 'react';
import { shallow } from 'enzyme';
import Item from '../';

describe('Item component', () => {
  it('should render snapshot', () => {
  const wrapper = shallow(<Item handleClick = { jest.fn() }>Do homework</Item>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call onDelete with expected value', () => {
    const mock = jest.fn();
    const wrapper = shallow(<Item handleClick={ mock }>Do homework    </Item>);
    
    wrapper.find('button').simulate('click', { target: { previousSibling: { innerHTML: "Do homework" }}});

    expect(mock).toBeCalledTimes(1);
    expect(mock).toBeCalledWith("Do homework");
  });
});
