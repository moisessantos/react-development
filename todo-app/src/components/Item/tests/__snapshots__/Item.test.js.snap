// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should render snapshot 1`] = `
<li
  className="list-group-item list-group-item-action"
>
  <span>
    Do homework
  </span>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
